### PR TITLE
Create a component release template.

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@amazon.com>

### Description

- Added `release_template.md` for each individual component's release template to make it a permalink.
- We already have a `_` template, so I called this one the same.
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: meta/templates/releases/release_template.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+Coming from [opensearch-build__REPLACE_ISSUE_NUMBER__](https://github.com/opensearch-project/opensearch-build/issues/__REPLACE_ISSUE_NUMBER__), release version 1.2. Please follow the following checklist.
+
+### Preparation
+
+- [ ] Assign this issue to a release owner.
+- [ ] Finalize scope and feature set and update [the Public Roadmap](https://github.com/orgs/opensearch-project/projects/1).
+- [ ] Create, update, triage and label all features and issues targeted for this release with `v__REPLACE_MAJOR_MINOR_PATCH__`.
+
+### CI/CD - _Ends Nov 2nd_
```
Comment (peternied) [inline meta/templates/releases/release_template.md:9]: Missed something that needs a replaced annotation, looks like at the top of the issue too.

**Review by peternied — APPROVED** [review]
Love getting this process even more codified, I've got additional thoughts on the release template, but lets save that for many tiny PRs

**Review by peternied — APPROVED** [review]

**Review by tianleh — APPROVED** [review]

File: meta/templates/releases/release_template.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+### Release
+
+- [ ] Complete [documentation](https://github.com/opensearch-project/documentation-website).
+- [ ] Gather, review and publish release notes.
+- [ ] Verify all issued labeled for this release are closed or labelled for the next release.
+
+### Post Release
+
+- [ ] Create [a release tag](https://github.com/opensearch-project/.github/blob/main/RELEASING.md#tagging).
+- [ ] [Suggest improvements](https://github.com/opensearch-project/opensearch-build/issues/new) to [this template](https://github.com/opensearch-project/opensearch-build/meta/templates/releases/release_template.md).
```
Comment (tmarkley) [inline meta/templates/releases/release_template.md:37]: This link is broken. It should be `https://github.com/opensearch-project/opensearch-build/blob/main/meta/templates/releases/release_template.md`, right?

Comment (dblock) [reply meta/templates/releases/release_template.md:37]: Yes, my bad. Care to fix? :)

Comment (tmarkley) [reply meta/templates/releases/release_template.md:37]: Yeah of course! Working on that now.

**Pull request merged**